src/Lecture2.hs

```diff
@@ -264,7 +344,16 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval _ (Lit x)      = Right x
+eval vars (Var v)   = case lookup v vars of
+                        Nothing -> Left (VariableNotFound v)
+                        Just x -> Right x
+eval vars (Add opl opr) = (eval vars opl) `plus` (eval vars opr)
+     where
+        plus :: Either EvalError Int -> Either EvalError Int -> Either EvalError Int
+        plus (Left x) _ = Left x
+        plus _ (Left x) = Left x
+        plus (Right x) (Right y) = Right (x + y)
```

Comment:
That's a great helper function! 👏🏻 
Once you learn about Monads, you'll be able to reduce this function to a single line ✂️